airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java

```diff
@@ -89,36 +94,61 @@ public static XminStatus getXminStatus(final JdbcDatabase database) throws SQLEx
     return new XminStatus()
         .withNumWraparound(result.get(NUM_WRAPAROUND_COL).asLong())
         .withXminXidValue(result.get(XMIN_XID_VALUE_COL).asLong())
-        .withXminRawValue(result.get(XMIN_RAW_VALUE_COL).asLong());
+        .withXminRawValue(result.get(XMIN_RAW_VALUE_COL).asLong())
+        .withVersion(2L)
+        .withStateType(StateType.XMIN);
   }
 
-  public static void logFullVacuumStatus(final JdbcDatabase database, final ConfiguredAirbyteCatalog catalog, final String quoteString) {
-    catalog.getStreams().forEach(stream -> {
+  public static Map<AirbyteStreamNameNamespacePair, Long> fileNode(final JdbcDatabase database, final List<ConfiguredAirbyteStream> streams, final String quoteString) {
+    final Map<AirbyteStreamNameNamespacePair, Long> fileNodes = new HashMap<>();
+    streams.forEach(stream -> {
+      final AirbyteStreamNameNamespacePair namespacePair = new AirbyteStreamNameNamespacePair(stream.getStream().getName(), stream.getStream().getNamespace());
+      final long l = fileNode(database, namespacePair, quoteString);
+      fileNodes.put(namespacePair, l);
+    });
+    return fileNodes;
+  }
+
+  public static long fileNode(final JdbcDatabase database, final AirbyteStreamNameNamespacePair stream, final String quoteString) {
```

Comment:
nit : can we name this something like fileNodeForStream and not make it a `public` method?